cmd/sdkch/main.go

```diff
@@ -44,22 +47,91 @@ var (
 		"sdk":        "SDK",
 		"tendermint": "Tendermint",
 	}
+
+	// RootCmd represents the base command when called without any subcommands
+	RootCmd = &cobra.Command{
+		Use:   "sdkch",
+		Short: "\nMaintain unreleased changelog entries in a modular fashion.",
+	}
+
+	// command to add a pending log entry
+	AddCmd = &cobra.Command{
+		Use:   "add [section] [stanza] [message]",
+		Short: "Add an entry file.",
+		Long: `
+Add an entry file. If message is empty, start the editor to edit the message.
+
+    Sections             Stanzas
+         ---                 ---
+    breaking                gaia
+    features             gaiacli
+improvements            gaiarest
+    bugfixes                 sdk
+                      tendermint`,
+		Args: cobra.MaximumNArgs(3),
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if interactiveMode {
+				addEntryFileFromConsoleInput()
+				return
+			}
+
+			if len(args) < 2 {
+				log.Println("must include at least 2 arguments when not in interactive mode")
+				return
+			}
+			sectionDir, stanzaDir := args[0], args[1]
+			validateSectionStanzaDirs(sectionDir, stanzaDir)
+			if len(args) == 3 {
+				addSinglelineEntryFile(sectionDir, stanzaDir, strings.TrimSpace(args[2]))
+				return
+			}
+			addEntryFile(sectionDir, stanzaDir)
+		},
+	}
+
+	// command to generate the changelog
+	GenerateCmd = &cobra.Command{
+		Use:   "generate",
+		Short: "Generate a changelog in Markdown format and print it to STDOUT. version defaults to UNRELEASED.",
+		Args:  cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			version := "UNRELEASED"
+			if flag.NArg() > 1 {
+				version = strings.Join(flag.Args()[1:], " ")
+			}
+			generateChangelog(version)
```

Comment:
I'd make downstream return an error, and then:

```suggestion
			return generateChangelog(version)
```

So that we can even avoid using `log.Fatal()`.
Ditto for all other command handlers.

Reply:
Cool I've implemented so that all (besides one in `init`) `log.Fatal`'s have been removed